src/main/java/duke/Commands/Command.java

```diff
@@ -0,0 +1,174 @@
+        int index = formatNumber(userInput);
+        if (index < 0 || index > listOfTasks.size() - 1) {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but the number you provided is invalid\n");
+        }
+        Task toUnmark = listOfTasks.get(index);
+        toUnmark.unmark();
+        StringBuilder toPrint = new StringBuilder("OK, I've marked this task as not done yet:\n")
+                .append(toUnmark)
+                .append("\n");
+        return toPrint.toString();
+    }
+
+    private static String findTask(String keyword, TaskList listOfTasks) {
+        StringBuilder toPrint = new StringBuilder("Here are the matching tasks in your list:\n");
+        int count = 1;
+        for (Task task : listOfTasks.getList()) {
+            if (task.describe().contains(keyword)) {
+                String toAdd = count + "." + task + "\n";
+                toPrint.append(toAdd);
+                count++;
+            }
+        }
+        return toPrint.toString();
+    }
+
+    /**
+     * Contains the logic for processing the processed userInput, finds the corresponding commands
+     * to the userInput and executes it.
+     *
+     * @param processedInput String array containing valuable information regarding the command
+     *                       to execute
+     * @param listOfTasks TaskList containing the current list of task
+     * @return String containing any responses to the user after execution of command
+     */
+    public static String execute(String[] processedInput, TaskList listOfTasks) {
+        String toPrint = "";
+        try {
+            switch (processedInput[0]) {
+            case "list":
+                toPrint = listTasks(listOfTasks);
+                break;
+            case "unmark":
+                toPrint = unmarkTask(processedInput[1], listOfTasks);
+                break;
+            case "mark":
+                toPrint = markTask(processedInput[1], listOfTasks);
+                break;
+            case "delete":
+                toPrint = deleteTask(processedInput[1], listOfTasks);
+                break;
+            case "deadline":
+                toPrint = addDeadline(processedInput[1], processedInput[2], listOfTasks);
+                break;
+            case "event":
+                toPrint = addEvent(processedInput[1], processedInput[2], listOfTasks);
+                break;
+            case "todo":
+                toPrint = addTodo(processedInput[1], listOfTasks);
+                break;
+            case "find":
+                toPrint = findTask(processedInput[1], listOfTasks);
+                break;
+            default:
+                throw new DukeException("☹ OOPS!!! I'm sorry, "
+                        + "but I don't know what that means :-(\n");
+            }
+        } catch (DukeException e) {
+            toPrint = e.getMessage();
+        }
+        return toPrint;
+    }
+
```

Comment:
You might want to consider writing some classes inherited from Command classes to abstract different methods, e.g. MarkCommand etc? It might help to enhance code readability 😉